src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -413,54 +406,66 @@ private void handleMetricsTag(InputFile inputFile) throws XMLStreamException {
       }
     }
 
+    private <T extends Serializable> void saveMetric(InputFile inputFile, Metric<T> metric, T value) {
+      context.<T>newMeasure()
+        .on(inputFile)
+        .forMetric(metric)
+        .withValue(value)
+        .save();
+    }
+
+    private void parseAndSaveIntegerMetric(InputFile inputFile, Metric<Integer> metric) throws XMLStreamException {
+      int value = Integer.parseInt(stream.getElementText());
+      saveMetric(inputFile, metric, value);
+    }
+
+    private void parseAndSaveStringMetric(InputFile inputFile, Metric<String> metric) throws XMLStreamException {
+      String value = stream.getElementText();
+      saveMetric(inputFile, metric, value);
+    }
+
     private void handleLinesMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.LINES, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.LINES);
     }
 
     private void handleClassesMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.CLASSES, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.CLASSES);
     }
 
     private void handleAccessorsMetricTag(InputFile inputFile) throws XMLStreamException {
-      double value = Double.parseDouble(stream.getElementText());
-      context.saveMeasure(inputFile, CoreMetrics.ACCESSORS, value);
+      parseAndSaveIntegerMetric(inputFile, CoreMetrics.ACCESSORS);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "ACCESSORS"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)